apps/server/src/modules/tool/school-external-tool/service/school-external-tool-metadata.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { ContextExternalToolRepo, SchoolExternalToolRepo } from '@shared/repo';
+import { ToolContextType } from '../../common/enum';
+import { SchoolExternalToolMetadata } from '../domain';
+
+@Injectable()
+export class SchoolExternalToolMetadataService {
+	constructor(
+		private readonly schoolToolRepo: SchoolExternalToolRepo,
+		private readonly contextToolRepo: ContextExternalToolRepo
+	) {}
+
+	async getMetadata(schoolExternalToolId: EntityId) {
```

Comment:
code duplication

Reply:
I don't see a way to remove the duplication. Just do the same refactoring as in external tools.